## Investigation

In [1]:
%run './lib/libraries.ipynb'

dython              0.6.1
tinydb              4.2.0


In [2]:
dfFlt  = load_df('dfFlt')

In [3]:
list_todo()


{'todo': 'Investigate 3298 null values in response_correctness', 'finished': False}
{'todo': 'Investigate 1566 [unassigned] in response_correctness', 'finished': False}
{'todo': 'Investigate null values in item_type_code_name', 'finished': False}
{'todo': 'Reduce number of levels in item_type_code_name', 'finished': False}
{'todo': 'Remove unary learner_attempt_status = "fully scored"', 'finished': False}
{'todo': 'Investigate number_of_learners > 40 outliers', 'finished': False}
{'todo': 'Investigate binary variables assignment_attempt_number and assignment_max_attempts', 'finished': False}
{'todo': 'Investigate assignment_attempt_number and assignment_max_attempts both have 1566 values', 'finished': False}
{'todo': 'Investigate final_score_unweighted has 16,582 zero values', 'finished': False}
{'todo': 'Investigate why some dates are null', 'finished': False}
{'todo': 'Remove variables assignment_late_submission, is_manual_scoring_required and is_deleted', 'finished': False}


### [Investigate binary variables assignment_attempt_number and assignment_max_attempts](./investigation/attempt-number-vs-max.ipynb)

### [Investigate 1566 [unassigned] in response_correctness](./investigation/1566-unassigned-in-response-correctness.ipynb)

### [Investigate 3298 null values in response_correctness](./investigation/3298-null-in-response-correctness.ipynb)

### [Investigate null values in item_type_code_name](./investigation/item-type-code-name-nulls.ipynb)

### [Investigate number_of_learners > 40 outliers](./investigation/num-learners-over-40.ipynb)

In [6]:
list_todo()

{'todo': 'Investigate 3298 null values in response_correctness', 'finished': True}
{'todo': 'Investigate 1566 [unassigned] in response_correctness', 'finished': True}
{'todo': 'Investigate null values in item_type_code_name', 'finished': True}
{'todo': 'Reduce number of levels in item_type_code_name', 'finished': False}
{'todo': 'Remove unary learner_attempt_status = "fully scored"', 'finished': False}
{'todo': 'Investigate number_of_learners > 40 outliers', 'finished': True}
{'todo': 'Investigate binary variables assignment_attempt_number and assignment_max_attempts', 'finished': True}
{'todo': 'Investigate assignment_attempt_number and assignment_max_attempts both have 1566 values', 'finished': True}
{'todo': 'Investigate final_score_unweighted has 16,582 zero values', 'finished': False}
{'todo': 'Investigate why some dates are null', 'finished': False}
{'todo': 'Remove variables assignment_late_submission, is_manual_scoring_required and is_deleted', 'finished': False}
{'todo': 'Impu

### [Investigate final_score_unweighted has 16,582 zero values](./investigation/final-score-unweighted-zero-values.ipynb)

### [Investigate 1566 null dates](./investigation/investigate-1566-null-dates.ipynb)

### [Investigate 749 to 750 null dates](./investigation/investigate-749-to-750-null-dates.ipynb)

### [Investigate 3422 null dates in scored_datetime](./investigation/investigate-3422-null dates-in-scored-datetime.ipynb)

### [Investigate 9965 null dates in was_in_progress_datetime](./investigation/investigate-9965-null-dates-in-was-in-progress-datetime.ipynb)

### [Investigate 18469 null dates in was_submitted_datetime_actual](./investigation/investigate-18469-null-dates-in was-submitted-datetime-actual.ipynb)

### [Investigate 15710 null assessment_item_response_id](./investigation/investigate-15710-null-assessment-item-response-id.ipynb)

###### Statistics of final_score_unweighted to item_type_code_name = Null

In [5]:
dfInv.filter(F.col("item_type_code_name").isNull() == True).select("final_score_unweighted").toPandas().hist(bins=60)

NameError: name 'dfInv' is not defined

- Strongly correlated to a final_score_unweighted of 0

In [ ]:
finish_todo('Investigate null values in item_type_code_name')


